### PR TITLE
[JBIDE-23029] allow integration tests to use core classes

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.common.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -27,69 +27,78 @@ Export-Package: org.jboss.tools.openshift.common.core;
    org.jboss.tools.openshift.express.test,
    org.jboss.tools.openshift.express.ui,
    org.jboss.tools.openshift.test,
-   org.jboss.tools.openshift.ui",
+   org.jboss.tools.openshift.ui,
+   org.jboss.tools.openshift.reddeer",
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.common.core/META-INF/MANIFEST.MF]: Why are you exporting reddeer packages in core ?

File: plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -20,6 +20,7 @@ Bundle-ClassPath: lib/openshift-restclient-java-5.1.0.Final.jar,
  lib/commons-io-2.1.jar,
  .
 Export-Package: com.openshift.internal.restclient;x-friends:="org.jboss.tools.openshift.test",
+ com.openshift.internal.restclient.authorization,
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: internal package should be exported to friends only, no ?

Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: +1

Comment (adietish) [inline plugins/org.jboss.tools.openshift.client/META-INF/MANIFEST.MF]: btw. for great reusability the jobs within openshift.ui should be living in core. Their design (dependencies) prevent that unfortunately. IMHO not the best design. That's not the moment to address this though.

**Comment by adietish** [comment]
@jeffmaury I'm not reexporting reddeer packages. I make sure common.core packages are only exported to friends, among them reddeer (which is questionable given they're not internal classes):
> @@ -27,69 +27,78 @@ Export-Package: org.jboss.tools.openshift.common.core;
>     org.jboss.tools.openshift.express.test,
>     org.jboss.tools.openshift.express.ui,
>     org.jboss.tools.openshift.test,
> -   org.jboss.tools.openshift.ui",
> +   org.jboss.tools.openshift.ui,
> +   org.jboss.tools.openshift.reddeer",

**Comment by adietish** [comment]
@jeffmaury removed the friends from public (non-internal) packages

**Pull request merged**